include/fastdds/rtps/messages/CDRMessage.hpp

```diff
@@ -235,15 +235,23 @@ inline SequenceNumberSet_t CDRMessage::readSequenceNumberSet(
     SequenceNumberSet_t sns(seqNum);
     uint32_t numBits = 0;
     valid &= CDRMessage::readUInt32(msg, &numBits);
-    uint32_t n_longs = (numBits + 31ul) / 32ul;
-    uint32_t bitmap[8];
-    for (uint32_t i = 0; i < n_longs; ++i)
+    if (numBits > 256u)
     {
-        valid &= CDRMessage::readUInt32(msg, &bitmap[i]);
+        sns.base(c_SequenceNumber_Unknown);
+        valid = false;
```

Comment:
We can make `valid` local to the `else` clause, as it is not used anywhere else.

Reply:
No. What we should do is to check it along with the check on the number of bits.

Reply:
Done in 590b690dc5d68e07cbefeea8ef6d17282d39c2af